server/src/transport/http.rs

```diff
@@ -0,0 +1,440 @@
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use futures_util::TryStreamExt;
+use http::Method;
+use jsonrpsee_core::error::GenericTransportError;
+use jsonrpsee_core::http_helpers::read_body;
+use jsonrpsee_core::logger::{self, HttpLogger as Logger};
+use jsonrpsee_core::server::helpers::{prepare_error, BatchResponse, BatchResponseBuilder, MethodResponse};
+use jsonrpsee_core::server::rpc_module::MethodKind;
+use jsonrpsee_core::server::{resource_limiting::Resources, rpc_module::Methods};
+use jsonrpsee_core::tracing::{rx_log_from_json, tx_log_from_str, RpcTracing};
+use jsonrpsee_core::JsonRawValue;
+use jsonrpsee_types::error::{ErrorCode, BATCHES_NOT_SUPPORTED_CODE, BATCHES_NOT_SUPPORTED_MSG};
+use jsonrpsee_types::{ErrorObject, Id, Notification, Params, Request};
+use tokio::sync::OwnedSemaphorePermit;
+use tracing_futures::Instrument;
+
+type Notif<'a> = Notification<'a, Option<&'a JsonRawValue>>;
+
+/// Checks that content type of received request is valid for JSON-RPC.
+pub(crate) fn content_type_is_json(request: &hyper::Request<hyper::Body>) -> bool {
+	is_json(request.headers().get("content-type"))
```

Comment:
are headers always normalised to be lowercase (ie would this work if somebody provided Content-Type)?

Reply:
yes, they are are normalized to lower-case but let's future proof and use the constants from hyper instead.

Reply:
Ooh even better!